From 0a25cae713162545abdfe7000f32ea0f424f9b85 Mon Sep 17 00:00:00 2001
From: Florian Westphal <fwestpha@redhat.com>
Date: Tue, 2 Sep 2014 10:15:20 -0400
Subject: [net] l2tp: Enable checksum unnecessary conversions for l2tp/UDP sockets

Message-id: <1409652920-431-58-git-send-email-fwestpha@redhat.com>
Patchwork-id: 90168
O-Subject: [PATCH 57/57] [RHEL7.1 net] l2tp: Enable checksum unnecessary conversions for l2tp/UDP sockets
Bugzilla: 1131999
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>
RH-Acked-by: Vlad Yasevich <vyasevic@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>
RH-Acked-by: Jiri Benc <jbenc@redhat.com>

BZ: 1131999

commit 72297c59f7188d12f24daee1877e7dcca787cf1f
Author: Tom Herbert <therbert@google.com>
Date:   Sun Aug 31 15:12:46 2014 -0700

    l2tp: Enable checksum unnecessary conversions for l2tp/UDP sockets

    Signed-off-by: Tom Herbert <therbert@google.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/l2tp/l2tp_core.c b/net/l2tp/l2tp_core.c
index 5f52799..4fb799a 100644
--- a/net/l2tp/l2tp_core.c
+++ b/net/l2tp/l2tp_core.c
@@ -1394,6 +1394,8 @@ static int l2tp_tunnel_sock_create(struct net *net,
   if (!cfg->use_udp_checksums)
    sock->sk->sk_no_check_tx = 1;
 
+  udp_set_convert_csum(sock->sk, true);
+
   break;
 
  case L2TP_ENCAPTYPE_IP:
-- 
1.7.1